cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -334,6 +353,12 @@ func (cm *certManagerClient) shouldUpgrade(objs []unstructured.Unstructured) (st
 		return "", "", false, errors.Wrapf(err, "failed to parse config version [%s] for cert-manager component", desiredVersion)
 	}
 
+	// If the number of avaiable objects and objects to install differ, return true, but first fetch the current version.
+	diffObjLen := false
+	if len(objs) != len(installObjs) {
+		diffObjLen = true
+	}
+
 	needUpgrade := false
 	currentVersion := ""
 	for i := range objs {
```

Comment:
Let's fix the comment in l.392 I think it's misleading (or at least very hard to understand)